sdks/java/io/file-schema-transform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileReadSchemaTransformConfiguration.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.fileschematransform;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+import com.google.auto.value.AutoValue;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Strings;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Sets;
+
+@SuppressWarnings({
+  "nullness" // TODO(https://github.com/apache/beam/issues/20497)
+})
```

Comment:
Is there a way to apply nullness warnings suppression at the methods that are giving the problem? It's more code but alternatively do you like the idea of using Optional?  In the configuration I don't imagine a performance hit as it is evaluated at pipeline construction time.

Reply:
I git cloned this PR and tried these changes and was able to get a successful `./gradlew :sdks:java:io:file-schema-transform:check`. Would you consider to incorporating these changes?  I also provided comments at the lines of code what changes I was able to make to remove the null warnings without using the SuppressWarnings annotation.  I agree that we are applying `checkArgument` or `checkState` and why doesn't the checker pass kinds of questions. However, I hope that the following list could expedite the refactoring of this PR. (Note: Optional.isPresent seems to do the trick)

1. Remove all SupressWarnings annotations
2. Add a package private `getSafeSchema` method:

```
// FYI these are the imports:
// import java.util.Optional;
// import org.checkerframework.checker.nullness.qual.NonNull;

 @NonNull
  String getSafeSchema() {
    Optional<String> schema = Optional.ofNullable(getSchema());
    checkState(schema.isPresent() && !schema.get().isEmpty());
    return schema.get();
  }
```

3. Add a package private `getSafeFilepattern()`:

```
// FYI these are the imports:
// import java.util.Optional;
// import org.checkerframework.checker.nullness.qual.NonNull;
  @NonNull
  String getSafeFilepattern() {
    Optional<String> value = Optional.ofNullable(getFilepattern());
    checkState(value.isPresent() && !value.get().isEmpty());
    return value.get();
  }
```

Reply:
Sorry this suggestion slipped my mind last review. Will implement this, thanks for the examples